services/venice-server/src/main/java/com/linkedin/venice/listener/StorageReadRequestsHandler.java

```diff
@@ -397,8 +399,11 @@ private ReadResponse handleSingleGetRequest(GetRouterRequest request) {
     response.setValueRecord(valueRecord);
 
     if (keyValueProfilingEnabled) {
-      response.setOptionalKeySizeList(Optional.of(Arrays.asList(key.length)));
-      response.setOptionalValueSizeList(Optional.of(Arrays.asList(response.isFound() ? valueRecord.getDataSize() : 0)));
+      IntList keyList = new IntArrayList(Collections.singletonList(key.length));
+      response.setKeySizeList(keyList);
```

Comment:
Can we replace this by:

```
response.setKeySizeList(IntLists.singleton(key.lenght));
```

And same for the value size list, below?

Using `Collections.singletonList` will do boxing, which defeats the purpose of using an `IntList`.